packages/p2p-v2/src/components/Modals/NicknameModal/NicknameModal.tsx

```diff
@@ -51,63 +45,63 @@ const NicknameModal = ({ isModalOpen, setIsModalOpen }: TNicknameModalProps) =>
+                    <Text className='p2p-v2-nickname-modal__body-title' weight='bold'>
+                        What’s your nickname?
+                    </Text>
+                    <Text align='center' className='mt-4 mb-6' size={textSize}>
+                        Others will see this on your profile, ads and charts.
+                    </Text>
+                    <Controller
+                        control={control}
+                        name='nickname'
+                        render={({ field: { onBlur, onChange, value }, fieldState: { error } }) => (
+                            <Input
+                                data-testid='dt_p2p_v2_nickname_modal_input'
+                                error={!!error?.message || isError}
+                                label='Your nickname'
+                                message={createError?.error?.message || error?.message}
+                                onBlur={onBlur}
+                                onChange={onChange}
+                                value={value}
+                            />
+                        )}
+                        rules={{
+                            pattern: {
+                                message: 'Can only contain letters, numbers, and special characters .-_@.',
+                                value: /^[a-zA-Z0-9.@_-]*$/,
+                            },
+                            required: 'Nickname is required',
+                        }}
+                    />
+                    <Text className='my-10' size={textSize}>
+                        Your nickname cannot be changed later.
+                    </Text>
+                </Modal.Body>
+                <Modal.Footer className='p-0 min-h-0' hideBorder>
                     <Button
-                        className='p2p-v2-nickname-modal__form__button-group__cancel'
+                        className='border-2 mr-[0.8rem]'
+                        color='black'
                         onClick={() => {
-                            switchTab('buy-sell');
+                            history.push(BUY_SELL_URL);
                             setIsModalOpen(false);
                         }}
                         size='lg'
+                        textSize={textSize}
                         type='button'
                         variant='outlined'
                     >
-                        <Text size={textSize} weight='bold'>
-                            Cancel
-                        </Text>
+                        Cancel
                     </Button>
-                    <Button disabled={!isValid || !isDirty || isError} size='lg' type='submit'>
-                        <Text color='white' size={textSize} weight='bold'>
-                            Confirm
-                        </Text>
+                    <Button disabled={!isValid || !isDirty || isError} size='lg' textSize={textSize} type='submit'>
```

Comment:
do we need to specify the type='submit' here? since it would default to submit type